packages/gestalt/src/Upsell.js

```diff
@@ -70,50 +77,89 @@ const UpsellLink = ({
 
 export default function Upsell({
   dismissButton,
+  imageData,
   message,
   primaryLink,
   secondaryLink,
   title,
 }: Props): Node {
+  const isImage = imageData && imageData.component.type === Image;
+
   return (
     <Box
       display="flex"
       direction="column"
-      mdDirection="row"
+      smDirection="row"
       padding={6}
-      mdPadding={8}
+      smPadding={8}
       position="relative"
       rounding={4}
       borderStyle="shadow"
     >
       <Box
-        mdDisplay="flex"
+        smDisplay="flex"
         wrap
         width="100%"
-        mdMarginTop={-3}
-        mdMarginBottom={-3}
+        smMarginTop={-3}
+        smMarginBottom={-3}
       >
         <Box
           display="flex"
+          direction="column"
+          smDirection="row"
+          justifyContent="center"
+          alignItems="center"
           marginBottom={primaryLink || secondaryLink ? 4 : undefined}
-          mdMarginBottom={primaryLink || secondaryLink ? 0 : undefined}
-          mdPaddingY={3}
+          smMarginBottom={primaryLink || secondaryLink ? 0 : undefined}
+          smPaddingY={3}
         >
-          <Box
-            marginBottom="auto"
-            marginTop="auto"
-            maxWidth={648}
-            marginEnd={6}
-          >
-            {title && (
-              <Box marginBottom={2}>
-                <Heading size="sm">{title}</Heading>
+          {imageData && (
+            <Box
+              marginBottom={4}
+              smMarginBottom={0}
+              width={isImage ? Math.min(imageData.width, 128) : undefined}
```

Comment:
can we do 

`width={isImage ? Math.min(imageData.width || 128, 128) : undefined}`

`Math.min` doesnt take `undefined`